arch/riscv/core/fatal.c

```diff
@@ -80,12 +80,7 @@ const char *z_riscv_mcause_str(unsigned long cause)
 FUNC_NORETURN void z_riscv_fatal_error(unsigned int reason,
```

Comment:
Consider making the `sp` always available in the `arch_esf` when `CONFIG_EXCEPTION_DEBUG=y` so that handlers do not have to deduce it by offsetting `*esf` everywhere